### ISSUE TITLE
[cli] Rancher 2.6 Project context switching support

### ISSUE THREAD

**Issue by vtrenton** [rancher/rancher#34723]
Hello It seems the CLI doesn't support project context switching within 2.6 clusters just as of yet.

running latest stable Rancher CLI version: v2.4.12

It looks like the cli doesn't like the new c-m-* as opposed to the older c-* project naming convention

```
$ rancher login -t <token> --context 'c-m-qvdxgtc2:p-bgvwr' https://<rancher cluster dns>

FATA[0002] Unable to parse context (c-m-qvdxgtc2:p-bgvwr). Please provide context as local:p-xxxxx, c-xxxxx:p-xxxxx, or c-xxxxx:project-xxxxx

$ rancher context switch
NUMBER    CLUSTER NAME      PROJECT ID             PROJECT NAME   PROJECT DESCRIPTION
1         testing-context   c-m-qvdxgtc2:p-bgvwr   System         System project created for the cluster
2         testing-context   c-m-qvdxgtc2:p-lwwf9   Default        Default project created for the cluster
3         local             local:p-f8s9m          Default        Default project created for the cluster
4         local             local:p-phx6w          System         System project created for the cluster
```

**Comment by dramich** [comment]
The regex needs to be updated to include `c-m` to basically match how project works. https://github.com/rancher/cli/blob/master/cmd/common.go#L536

**Comment by anupama2501** [comment]
Reproduced this issue on v2.6.0

- Works for rke1 clusters and local cluster without any errors but not for rke2 clusters which have the cluster name that start with `c-m`.
- For rke2 clusters, the following error is thrown:

`
FATA[0002] Unable to parse context (c-m-jlmjq882:p-hp9hn). Please provide context as local:p-xxxxx, c-xxxxx:p-xxxxx, or c-xxxxx:project-xxxx
`

**To reproduce:**

1. Create an rke1 node driver and rke2 node driver cluster or custom clusters.
2. Download rancher cli and login to the rancher with the command
./rancher login https://<redacted> --token token-<redacted> --context 'c-m-ztq8qwcf:p-xb8gw'
3. Error is thrown as mentioned above.
4. As a workaround, if we try logging in and then select the context it works:
`./rancher login https://<redacted> --token token-<redacted>`
`./rancher context switch` --> select the project number to switch to
```
INFO[0003] Setting new context to project System
INFO[0003] Saving config to /Users/anupamaupadhyayula/.rancher/cli2.json
```

**Comment by StrongMonkey** [comment]
Should be available to test when v2.4.13-rc1 is ready. https://github.com/rancher/cli/releases/tag/v2.4.13-rc1

**Comment by anupama2501** [comment]
Verified with the rancher cli version: v2.4.13-rc1, rancher server: v2.6-head 1a57856

1. Created a downstream rke2 and rke1 node driver clusters, custom clusters 
2. from the rancher cli gave the command
`./rancher login -t token-redacted --context 'c-m-z6thzkwx:p-24p5m' https://redacted`
3. Works as expected.
4.  Verified for rke1 clusters and local clusters - works as expected
